File: Data Structures & Algorithms/is-palindrome/submission-0.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int right = s.length()-1;
        int left = 0;
        while(left<right){
            if(!isalnum(s[left])){
                left++;
                continue;
            }
            if(!isalnum(s[right])){
                right--;
                continue;
            }
            if(tolower(s[left])==tolower(s[right])){
                left++;
                right--;
            }
            else{
                return false;
            }
        }
        return true;
    }
};

File: Data Structures & Algorithms/reverse-string/submission-0.py
class Solution(object):
    def reverseString(self, s):
        right = len(s)-1
        left = 0
        while left<right:
            temp = s[left]
            s[left] = s[right]
            s[right] = temp
            right-=1
            left+=1
        return s
